enhancements/network/allow-from-router-networkpolicy.md

```diff
@@ -0,0 +1,236 @@
+ingress policies work correctly, while the primary node IP would be
+needed to make pod-to-node egress policies work correctly.)
+
+The question then is what namespace this should be; I don't think it
+should be "`default`" like in openshift-sdn, since that namespace is
+not actually supposed to be "magic" in this way. In OCP, we'll want
+the "host-network-matching" namespace to be something starting with
+"`openshift-`", but we can't hardcode that upstream.
+
+So the best solution seems to be that we add a config option
+`host-network-namespace` to the `[kubernetes]` section of the config.
+If this option is specified, and the indicated namespace exists, then
+ovn-kubernetes will do whatever it is that it has to do, to make it so
+that NetworkPolicies that select that namespace get applied to
+host-network traffic.
+
+So, then, the plan is:
+
+  1. When using ovn-kubernetes, CNO will create a namespace
+     `openshift-host-network`, and configure ovn-kubernetes in OCP
+     with `host-network-namespace=openshift-host-network`.
+
+  2. For consistency, we modify openshift-sdn to treat
+     `openshift-host-network` the same way (see below for more
+     details).
+
+  3. CNO will watch the ingress configuration, and when the endpoint
+     publishing strategy is `HostNetwork`, it will add the
+     `network.openshift.io/policy-group: ingress` label to
+     `openshift-host-network`, and when it's not `HostNetwork` it will
+     remove that label.
+
+Then users will be able to match router traffic in either
+openshift-sdn or ovn-kubernetes with any router endpoint publishing
+strategy via:
+
+    ingress:
+      - from:
+          - namespaceSelector:
+              matchLabels:
+                network.openshift.io/policy-group: ingress
+
+We can also formally specify "allow-from-host-network". (This is
+basically free at this point, and also useful to users, so...) We just
+have CNO also add the label `network.openshift.io/policy-group:
+host-network` to `openshift-host-network`, and then we document that
+(at least when using openshift-sdn or ovn-kubernetes), a NetworkPolicy
+selecting namespaces with that label will match host-network traffic.
+
+#### OpenShift SDN Modifications
+
+We _could_ avoid modifying openshift-sdn, and just have CNO know that
+the "host network" namespace is `openshift-host-network` with
+ovn-kubernetes, but `default` with openshift-sdn. However, it would be
+simpler if we just always used the same namespace as the "host network
+namespace".
+
+In NetworkPolicy mode, openshift-sdn doesn't let you have two
+namespaces with the same VNID, but we could easily hack it so that
+`openshift-host-network` was treated as VNID 0 _for purposes of
+NetworkPolicy only_. This would cause it to behave strangely if there
+were any pods in it, but we don't really care, since there _won't_ be
+any pods in it.
+
+### Risks and Mitigations
+
+The biggest problem is that there is really no good way to distinguish
```

Comment:
Is traffic from host-network router using a specific subset of the ephemeral port range? If not, can we add a parameter to restrict what traffic (belonging to what port range) is treated as from router and have the router pods only use those ports?

Reply:
NetworkPolicies cannot filter on source port.